ieeeuoft/hackathon-template
# IEEE-132 and IEEE-133: Add cancelOrderThunk and connect to Order Tables

## Overview

- Resolves #IEEE-132 and #IEEE-133
- Create cancelOrderThunk in orderSlice.ts to change order status to "Cancelled"
  - call PATCH request to endpoint: /api/hardware/orders/<id>
  - use removeOne from EntityAdapter
- Connect thunk to Order Tables
  - when cancel button is clicked, it dispatches thunk


## Unit Tests Created

- Tested state of cancelOrderLoadingSelector
- Mocked API Response for cancelOrderThunk and ensured pendingOrderAdapter removed the order 
- Tested error and success snack bars
- Tested is cancel order button is hidden when 'hide all' is selected
- Tested UI when cancel order button is clicked


## Steps to QA

- Add cart items, and submit them as an order. Then go to Dashboard and click 'Cancel Order' to see the orders get removed.
